File: 4-Find-Longest-Word.js
// ====== My Solution ======
function findLongestWord(str) {
    let wordArray = str.split(' ')
    let longestWord = wordArray[0]
    wordArray.forEach(word => {
      if (word.length > longestWord.length) {
        longestWord = word
      }
    })
    return longestWord.length
  }
  
  
  // ====== Using reduce ======
  function findLongestWordReduce(str) {
    return str.split(' ')
      .reduce((acc, curr) => {
        return Math.max(acc, curr.length)
      }, 0)
  }

File: 4-Find-Longest-Word.py
# using list comprehension
def findLongestWord(string):
    word_array = string.split()  # uses whitespace as default
    lengths_array = [len(word) for word in word_array]
    return max(lengths_array)

# in one line
def findLongestWord(string):
    return max([len(word) for word in string.split()])


# === tests ===
print(findLongestWord("The quick brown fox"))